gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala

```diff
@@ -433,8 +433,12 @@ case class AddTransformHintRule() extends Rule[SparkPlan] {
         case plan: FilterExec =>
           val childIsScan = plan.child.isInstanceOf[FileSourceScanExec] ||
             plan.child.isInstanceOf[BatchScanExec]
-          // When scanOnly is enabled, filter after scan will be offloaded.
-          if ((!scanOnly && !enableColumnarFilter) || (scanOnly && !childIsScan)) {
+          if (!enableColumnarFilter) {
```

Comment:
Why remove `!scanOnly`? The logic is different with before

Reply:
Yes, I think we should prioritize `enableColumnarFilter`

Reply:
I think we should fallback FilterExec when we enable `scanOnly` and disable `columnar filter`. what you think?

Reply:
@rui-mo